client/mysqlbinlog.cc

```diff
@@ -118,6 +119,9 @@ static enum_base64_output_mode opt_base64_output_mode= BASE64_OUTPUT_UNSPEC;
 static char *opt_base64_output_mode_str= NullS;
 static char* database= 0;
 static char* table= 0;
+static std::set<std::string> filter_databases;
+static std::set<std::string> filter_tables;
```

Comment:
Ayo, we usin’ C++ string and (**tree**)set instead of our `String` and (C-compatible?) `HASH` now? 😀 Count me in.

By the way, we don’t need the elements sorted, right? So how about?
```suggestion
static std::unordered_set<std::string> filter_databases;
static std::unordered_set<std::string> filter_tables;
```
https://en.cppreference.com/w/cpp/container/unordered_set

Also, while neither set compares `char*` _contents_ out-of-the-box (they only compare the `*`), we can optionally supply our own comparators in the template arguments, and even a memory allocator (`my_malloc`) too!

Reply:
A good point that the unordered-ness. I don't see either where the ordered access is required.